assets/js/googlesitekit/widgets/util/combine-widgets.js

```diff
@@ -21,11 +21,62 @@
  */
 import isEqual from 'lodash/isEqual';
 
+/**
+ * Internal dependencies
+ */
+import { SPECIAL_WIDGET_STATES } from './constants';
+
 function stateAndRowMatch( stateA, stateB, rowA, rowB ) {
 	return rowA === rowB && isEqual( stateA, stateB );
 }
 
 /**
+ * Determines whether all the widgets in a given area should be combined.
+ *
+ * The given widgets in an area can all be combined if they are
+ * all from the same module and they are all in the same special
+ * state (see #3225).
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<Object>} widgets      List of widgets.
+ * @param {Object}         widgetStates Map of widget slug and their
+ *                                             state (either an object with
+ *                                             `Component` and `metadata`, or
+ *                                             `null`).
+ * @return {boolean} 			Whether all the widgets should be combined.
+ */
+function shouldCombineAllWidgets( widgets, widgetStates ) {
+	const states = {};
+
+	for ( let i = 0; i < widgets.length; i++ ) {
+		const widget = widgets[ i ];
+		const widgetState = widgetStates?.[widget.slug];
+
+		const state = widgetState?.Component?.name;
+		const module = widgetState?.metadata?.moduleSlug;
+		const isSpecialState = SPECIAL_WIDGET_STATES.includes( state );
+		const isMultipleModules = Object.keys( states ).length > 1;
```

Comment:
It won't return `false` if the last widget is from another module: `[ analytics-widget-A, analytics-widget-B, search-console-widget-A ]`

Reply:
Great catch! Thank you.